### PR TITLE
[fmiweather] Add support for edited Scandinavia forecast

### LINKED ISSUE #17548: [fmiweather] Add possibility to use fmi::forecast::edited in addition to fmi::forecast::harmonie
# Problem / Motivation
The fmiweather binding fetches weather forecast from the Finnish Meteorology Instititute's (FMI) API. The binding uses currently the fmi::forecast::harmonie::surface::point::multipointcoverage stored query. The forecast provided as a response is based on the Harmonie weather model, which is different to what FMI publishes as their official forecast on their website and in their mobile app.

The meteorologists at FMI use multiple different weather models and their experience to provide an official FMI forecast which they publish on their website and in the FMI mobile app. [This official forecast, which is often times more accurate than the Harmonie-based forecast, has been available via their API since 6 June 2023](https://en.ilmatieteenlaitos.fi/open-data-changelog). During the past two winters, there have been cases where the Harmonie based forecast has been off by almost 10 degrees while the forecast provided by the FMI meteorologists have been accurate.

Having as accurate weather forecast as possible is cruicial for optimizing the heating of a house using openHAB. There are solutions like the [openHAB Spot Price Optimizer](https://community.openhab.org/t/spot-price-optimizer-advanced-algorhitms-to-optimize-heating-charging-of-electric-vehicles-water-boilers-and-more/159539) to do this (I'm the maintainer of that solution). At the moment the Spot Price Optimizer doesn't use the FMI Binding, but has it's own javascript module to fetch the FMI forecast.

# Proposed change
I would like to see this official FMI Weather Binding to allow users to be able to choose between the fmi::forecast::edited model and the current fmi::forecast::harmonie model.

### PR DESCRIPTION
**Pull request by jlaur**
Resolves #17548

### CODE REVIEW

File: bundles/org.openhab.binding.fmiweather/README.md
Comment (ssalonen) [inline bundles/org.openhab.binding.fmiweather/README.md:1]: Please mention this in the intro paragraph, we could shortly introduce this better forecast model

**Review by ssalonen — COMMENTED** [review]
Thank you once again! Great improvement

small comment on readme

File: bundles/org.openhab.binding.fmiweather/README.md
Comment (jlaur) [reply bundles/org.openhab.binding.fmiweather/README.md:1]: Perhaps you or @masipila can help me out a bit here? I only know that it's a different forecast edited by meteorologists, but I'm not into the details.

I can add that when testing with my own location, the two forecasts were exactly the same. When I changed to Helsinki, I could see differences between them.

Comment (masipila) [reply bundles/org.openhab.binding.fmiweather/README.md:1]: @jlaur and @ssalonen: How do you like this wordsmithing to the README.

The `forecast` Thing has a configuration option to choose between two different queries. Harmonie is one of the weather models that meteorologists use in their work. The `edited` query provides the official FMI forecast, which is often times more accurate since it's edited by meteorologists who combine several different weather models and their experience to produce the official forecast.

Comment (jlaur) [reply bundles/org.openhab.binding.fmiweather/README.md:1]: @ssalonen - I see now that HARMONIE was mentioned explicitly in the second sentence in the README, so this is probably what you meant by introducing the model in the intro paragraph. I have taken @masipila's suggestion (thanks!) and tried to fit it into this paragraph. How does it look to you now?

File: bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/ForecastWeatherHandler.java
Comment (lolodomo) [inline bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/ForecastWeatherHandler.java:101]: No need for String.format ... as there is nothing to format ;)

**Review by lolodomo — CHANGES_REQUESTED** [review]
Just one minor comment

File: bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/ForecastWeatherHandler.java
Comment (jlaur) [reply bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/ForecastWeatherHandler.java]: Thanks, I didn't notice when simplifying the `initialize` method. 🙂 Fixed, and also removed redundant logging.

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

File: bundles/org.openhab.binding.fmiweather/README.md
Comment (lolodomo) [reply bundles/org.openhab.binding.fmiweather/README.md:1]: This looks good to me. Please explain what is wrong for you @ssalonen. I am waiting for your feedback before I merge.

Comment (ssalonen) [reply bundles/org.openhab.binding.fmiweather/README.md:1]: I think you got it just right, thanks!

**Pull request merged**
